### PR TITLE
[Nightly test] Add a new env to fix issues

### PR DESCRIPTION
**Pull request by rkooo567**
## Why are these changes needed?

It is advised to add this env var to make k8s tests work

## Checks

- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: release/ray_release/cluster_manager/cluster_manager.py [hunk 1]
```diff
@@ -41,7 +41,7 @@ def set_cluster_env(self, cluster_env: Dict[str, Any]):
         self.cluster_env["env_vars"]["MATCH_AUTOSCALER_AND_RAY_IMAGES"] = "1"
         self.cluster_env["env_vars"]["RAY_gcs_storage"] = "memory"
         self.cluster_env["env_vars"]["RAY_USAGE_STATS_ENABLED"] = "1"
-        self.cluster_env["env_vars"]["RAY_USAGE_STATS_SOURCE"] = "nightly-tests"
+        self.cluster_env["env_vars"]["RAY_bootstrap_with_gcs"] = "1"
```
Comment (jjyao) [inline release/ray_release/cluster_manager/cluster_manager.py:44]: Why do we delete the SOURCE env var?

Comment (rkooo567) [reply release/ray_release/cluster_manager/cluster_manager.py:44]: Oh just a mistake. I am still experimenting with this pr. I will get back to you once i confirm it works

Comment (krfricke) [reply release/ray_release/cluster_manager/cluster_manager.py:44]: There's another subtle bug here (the hash is calculated without the env vars), I'm fixing this in the other PR (or you can add it here)

Comment (krfricke) [reply release/ray_release/cluster_manager/cluster_manager.py:44]: This one: https://github.com/ray-project/ray/pull/23277/files#diff-569103799258d2583bc74456ccdb07d9b3a7b3bf8673ad9b153cd37b633a8712L49
(feel free to add to this PR as well)

**Pull request closed without merging**